modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/api/construction/ConstructionPayloadsSuite.scala

```diff
@@ -0,0 +1,150 @@
+        |      "metadata":{
+        |        "producer":"0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5"
+        |      }
+        |    }
+        |  },
+        |  "operations":[
+        |    {
+        |      "operation_identifier":{
+        |        "index":5,
+        |        "network_index":0
+        |      },
+        |      "related_operations":[
+        |        {
+        |          "index":1
+        |        },
+        |        {
+        |          "index":2
+        |        }
+        |      ],
+        |      "type":"Transfer",
+        |      "status":"Accepted",
+        |      "account":{
+        |        "address":"DAG77VVVRvdZiYxZ2hCtkHz68h85ApT5b2xzdTkn",
+        |        "sub_account":{
+        |          "address":"DAG77VVVRvdZiYxZ2hCtkHz68h85ApT5b2xzdTkn",
+        |          "metadata":{}
+        |        },
+        |        "metadata":{}
+        |      },
+        |      "amount":{
+        |        "value":"1238089899992",
+        |        "currency":{
+        |          "symbol":"DAG",
+        |          "decimals":8,
+        |          "metadata":{
+        |            "Issuer":"Satoshi"
+        |          }
+        |        },
+        |        "metadata":{}
+        |      },
+        |      "coin_change":{
+        |        "coin_identifier":{
+        |          "identifier":"0x2f23fd8cca835af21f3ac375bac601f97ead75f2e79143bdf71fe2c4be043e8f:1"
+        |        },
+        |        "coin_action":"coin_created"
+        |      },
+        |      "metadata":{
+        |        "asm":"304502201fd8abb11443f8b1b9a04e0495e0543d05611473a790c8939f089d073f90509a022100f4677825136605d732e2126d09a2d38c20c75946cd9fc239c0497e84c634e3dd01 03301a8259a12e35694cc22ebc45fee635f4993064190f6ce96e7fb19a03bb6be2",
+        |        "hex":"48304502201fd8abb11443f8b1b9a04e0495e0543d05611473a790c8939f089d073f90509a022100f4677825136605d732e2126d09a2d38c20c75946cd9fc239c0497e84c634e3dd012103301a8259a12e35694cc22ebc45fee635f4993064190f6ce96e7fb19a03bb6be2"
+        |      }
+        |    }
+        |  ],
+        |  "metadata":{},
+        |  "public_keys":[
+        |    {
+        |      "hex_bytes":"string",
+        |      "curve_type":"secp256k1"
+        |    }
+        |  ]
+        |}""".stripMargin
+
+    // TODO how does one instantiate a BlockchainId?
+
+    for {
+      jsonStr <- IO.delay(sampleJson)
+      decoded = decode[ConstructionPayloads.Request](jsonStr)
+
+    } yield assert(decoded.isRight) // TODO compare actual decoded object?
+  }
+
+  test("encode any response") { _ =>
+    forall(responseGen) { response =>
+      for {
+        expected <- expectedJsonResponse(response)
+        actual = response.asJson.deepDropNullValues
+        _ = println(actual.spaces2)
+      } yield expect.same(expected, actual)
+    }
+  }
+
+  // Sample Response JSON based on
+  // https://www.rosetta-api.org/docs/ConstructionApi.html#constructionpayloads
+  // but with empty metadata fields removed
```

Comment:
imo you should be able to send empty metadata. our rosetta implementation should just ignore it

Reply:
@sdavidp My comment is to point out that I removed the empty metadata elements from the _response_, not the request. I thought I read the response shouldn't have null or empty objects.

Reply:
@marcinwadon  (re-typing this response since I accidentally tagged myself.)

My comment is to point out that I removed the empty metadata elements from the response, not the request. I thought I read the response shouldn't have null or empty objects before later PR comments confirmed that.